[31기 구본희] Add : 상세페이지 구조작업

bonirun opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 상세페이지 기능구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 스크롤 네비게이션 라이브러리 적용
2. 이미지 갤러리 기능구현 - React slick 라이브러리 사용
3. 찜하기 버튼 토글 기능 구현


<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 스타일컴포터넌트 사용법을 알게됬습니다.
- 라이브러리 적용 하는법을 배웠습니다.

<br />

## :: 기타 질문 및 특이 사항

yeonuk-hwang commented on package.json:
```diff
@@ -6,10 +6,15 @@
     "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^12.1.4",
     "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.26.1",
```
사용하지 않는 패키지 같네요 삭제해주세요!

yeonuk-hwang commented on src/pages/LectureDetail/Header.js:
```diff
@@ -0,0 +1,52 @@
+  font-weight: bold;
+  letter-spacing: -0.45px;
+  word-break: break-all;
+  color: ${props => props.theme.fontMain};
```
```suggestion
  color: ${({theme}) => theme.fontMain};
```

yeonuk-hwang commented on src/pages/LectureDetail/ImageViewer.js:
```diff
@@ -0,0 +1,163 @@
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+
+const images = [
```
상수데이터는 UPPERCASE + snake_case로 네이밍 지어주세요!

yeonuk-hwang replied:
```suggestion
const IMAGES = [
```

yeonuk-hwang commented on src/pages/LectureDetail/ImageViewer.js:
```diff
@@ -0,0 +1,163 @@
+];
+const ImageViewer = () => {
+  const [nav1, setNav1] = useState();
+  const [nav2, setNav2] = useState();
```
nav1, nav2는 어떤 의미가 있는지 잘 드러나지 않네요! 의미가 드러나게 네이밍 해주세요!

yeonuk-hwang commented on src/pages/LectureDetail/LectureDetail.js:
```diff
@@ -1,5 +1,164 @@
+      <Wrapper>
+        <Left>
+          <Header />
+          <div id="gallery">
```
- `id` 보다는 `className` 을 활용해주시기 바랍니다.
- HTML에서  `id` 는 문서 내에서 유일해야 합니다. 이러한 이유로 단일 요소를 다루는 경우 `id`를 사용할 수 있습니다.
- 하지만 정말로 해당 요소에 대한 이벤트나 스타일 핸들링이 프로젝트 전체에서 단 한번만 사용 되는지 충분히 고민해야합니다. 일반적으로 컴포넌트는 재사용 될 수 있기 때문에 같은 요소가 여러번 생성될 수 있습니다. 이러한 이유로 보통의 경우에는 `className` 을 사용합니다.

yeonuk-hwang commented on src/pages/LectureDetail/LectureDetail.js:
```diff
@@ -1,5 +1,164 @@
+  letter-spacing: -0.22px;
+  word-break: break-all;
+
+  section:first-child & {
```
이렇게 네스팅해서 태그를 찍는것보다는 section 자체를 스타일드 컴포넌트를 처리하는게 좋겠네요!

yeonuk-hwang commented on src/pages/LectureDetail/StickyNav.js:
```diff
@@ -0,0 +1,255 @@
+    };
+  }, []);
+
+  const [likeChanged, setLikeChanged] = useState(false);
```
- state 선언부가 이를 이용하는 함수보다 아래에 있다보니 조금 코드 읽기가 혼란스럽네요!! addWish 함수보다 상단에 있는게 좋아보입니다

yeonuk-hwang reviewed (changes requested):
고생하셨습니다 본희님! 리뷰 확인 및 반영해주세요!

yeonuk-hwang commented on src/pages/LectureDetail/ImageViewer.js:
```diff
@@ -0,0 +1,153 @@
+const ImageViewer = ({ images }) => {
+  const [viewer, setViewer] = useState();
+  const [thumbnail, setThumbnail] = useState();
+  return (
```
다른 문맥들끼리는 공백을 통해서 가독성에 신경써주세요

yeonuk-hwang commented on src/pages/LectureDetail/StickyNav.js:
```diff
@@ -0,0 +1,230 @@
+
+  const addToWish = event => {
+    event.preventDefault();
+    if (localStorage.getItem('Authorization') === null) {
+      alert('로그인부터 해라 뿌잉~ㅠ');
```
- localStorage.getItem, setItem 하는 코드들을 별도의 함수로 추출해서 공통적으로 활용하면 좋겠네요!
- 이렇게 코드가 그대로 노출되면 문제점은 일종의 중복이 발생합니다. JWT가 필요한 모든 코드에서 getItem("Authorization")을 할텐데, 만약 setItem하는 key값이 변경된다면 동시다발적으로 수정할 소요가 많아집니다.
- 하지만 별도의 getToken등의 함수로 추출해두고 해당 함수를 호출하는 식으로 사용한다면  getToken 함수 하나에서만 변경하면 되기에 변경에 유연하게 대응할 수 있게 됩니다.
- 마찬가지의 이유로 setItem 하는 부분도 추출해서 사용하면 중복을 없앨 수 있고, getToken과 setToken 함수가 한 파일에 붙어 있으면 서로 어떤 key값으로 set하고 get하는지 확인할 수 있기에 편리하고 안전하게 수정을 할 수 있습니다!

yeonuk-hwang reviewed (commented):
- 본희님 리뷰 확인하시고 수정하면 수정했다고 표시해주세요(코멘트, 이모지 등)
- 리뷰 드린 사항중에 반영 안된부분있습니다 꼼꼼히 확인해주세요!
- 새로 드린 리뷰 확인 및 반영해주세요! 특히 localStorage에서 token 관련한 리뷰내용은 팀원들과 다 함께 이해하고 공유해주세요!

Commit:
MODIFY : 충돌해결

MODIFY : 리뷰섹션  레이아웃 수정

MODIFY : 충돌해결

MODIFY : 충돌해결

MODIFY : 리뷰섹션  레이아웃 수정

MODIFY : 찜추가/제거 로직 구현

MODIFY : 찜기능 수정중

MODIFY : 컨플릭해결

MODIFY : 경로수정

MODIFY : 결제창 위치값 조정

MODIFY : 상세페이지 찜기능 적용

MODIFY : npm install

MODIFY : sider 수정

rm cache
